ompi/mca/osc/rdma/osc_rdma.h

```diff
@@ -673,4 +673,84 @@ static inline mca_btl_base_am_rdma_module_t *ompi_osc_rdma_selected_am_rdma(ompi
     return module->alternate_am_rdmas[btl_index];
 }
 
+/**
+ * @brief Copies from src to dest buffers that may or may not be accelerator
+ *        buffers
+ *
+ * @param[in]  dest           destination buffer
+ * @param[in]  src            source buffer
+ * @param[in]  size           length to copy
+ *
+ * @returns dest
+ */
+static inline void *osc_rdma_accelerator_mem_copy(void *dest, const void *src, size_t size)
+{
+    int res, dev_id;
+    uint64_t flags;
+    int dest_is_accel = opal_accelerator.check_addr(dest, &dev_id, &flags);
+    int src_is_accel = opal_accelerator.check_addr(src, &dev_id, &flags);
+
+    if (dest_is_accel == 0 && src_is_accel == 0) {
+        return memcpy(dest, src, size);
+    }
+
+    res = opal_accelerator.mem_copy(MCA_ACCELERATOR_NO_DEVICE_ID, MCA_ACCELERATOR_NO_DEVICE_ID,
+                                  dest, src, size, MCA_ACCELERATOR_TRANSFER_UNSPEC);
+    if (OPAL_SUCCESS != res) {
+        opal_output(0, "Error in accelerator memcpy");
+        abort();
```

Comment:
Do we really need to invoke here abort()? I would have assumed that indicating an error code is right choice and let the error handler take care of dealing with the error.

Reply:
I'll try to bubble up the error